static/app/views/issueDetails/correlatedIssues.tsx

```diff
@@ -0,0 +1,131 @@
+import type {TimePeriodType} from 'sentry/views/alerts/rules/metric/details/constants';
+import RelatedIssues from 'sentry/views/alerts/rules/metric/details/relatedIssues';
+import {Dataset, TimePeriod} from 'sentry/views/alerts/rules/metric/types';
+import {extractEventTypeFilterFromRule} from 'sentry/views/alerts/rules/metric/utils/getEventTypeFilter';
+import {isCrashFreeAlert} from 'sentry/views/alerts/rules/metric/utils/isCrashFreeAlert';
+import {fetchAlertRule} from 'sentry/views/alerts/utils/apiCalls';
+import {SectionKey} from 'sentry/views/issueDetails/streamline/context';
+import {InterimSection} from 'sentry/views/issueDetails/streamline/interimSection';
+
+interface CorrelatedIssuesProps {
+  event: Event;
+  group: Group;
+  organization: Organization;
+  project: Project;
+}
+
+export default function CorrelatedIssues({
+  organization,
+  event,
+  group,
+  project,
+}: CorrelatedIssuesProps) {
+  const [rule, setRule] = useState<any>(null);
+  const ruleId = event.contexts?.metric_alert?.alert_rule_id;
+  useEffect(() => {
+    async function getRuleData() {
+      if (!ruleId) {
+        return;
+      }
+      const ruleData = await fetchAlertRule(organization.slug, ruleId, {
+        expand: 'latestIncident',
+      });
+      setRule(ruleData);
+    }
+    getRuleData();
+  }, [ruleId, organization.slug]);
+
+  const openPeriod = group.openPeriods?.[0];
+  if (!ruleId || !openPeriod) {
+    return null;
+  }
+
+  const start = openPeriod.start;
+  let end = openPeriod.end;
+  if (!end) {
+    end = moment().toISOString();
+  }
+  const getTimePeriod = (): TimePeriodType => {
+    return {
+      start,
+      end,
+      period: TimePeriod.SEVEN_DAYS,
+      usingPeriod: false,
+      label: t('Custom time'),
+      display: (
+        <Fragment>
+          <DateTime date={moment.utc(start)} />
+          {' — '}
+          <DateTime date={moment.utc(end)} />
+        </Fragment>
+      ),
+      custom: true,
+    };
+  };
+
+  const timePeriod = getTimePeriod();
+
+  if (!rule || !timePeriod) {
+    return null;
+  }
+
+  const {dataset, query} = rule;
+
+  const queryParams = {
+    start,
+    end,
+    groupStatsPeriod: 'auto',
+    limit: 5,
+    ...(rule.environment ? {environment: rule.environment} : {}),
+    sort: rule.aggregate === 'count_unique(user)' ? 'user' : 'freq',
+    query,
+    project: [project.id],
+  };
+  const issueSearch = {
+    pathname: `/organizations/${organization.slug}/issues/`,
+    query: queryParams,
+  };
+
+  const actions = (
+    <LinkButton data-test-id="issues-open" size="xs" to={issueSearch}>
+      {t('Open in Issues')}
+    </LinkButton>
+  );
+
+  return (
+    <InterimSection
+      title={t('Correlated Issues')}
+      type={SectionKey.CORRELATED_ISSUES}
+      help={t('A list of issues that are correlated with this event')}
+      actions={actions}
+    >
+      {[Dataset.METRICS, Dataset.SESSIONS, Dataset.ERRORS].includes(dataset) && (
```

Comment:
it might make sense to check this earlier and return null instead of a blank section